code/modules/atmospherics/machinery/pipes/pipes.dm

```diff
@@ -127,3 +127,6 @@
 		pipe_color = paint_color
 		update_node_icon()
 	return paintable
+
+/obj/machinery/atmospherics/pipe/update_layer()
+	layer = initial(layer) + (piping_layer - PIPING_LAYER_DEFAULT) * PIPING_LAYER_LCHANGE + (GLOB.pipe_colors_ordered[pipe_color] * 0.0001)
```

Comment:
This is a nice effect. thanks
how many colors would we need to add to start getting conflicts?

Reply:
a lot more actually, since the list goes both positive and negative, i'd say that even with 100 colors we should be fine